packages/editor-ui/src/components/RunData.vue

```diff
@@ -1199,47 +1167,59 @@ export default defineComponent({
 			const itemsLabel = itemsCount > 0 ? ` (${itemsCount} ${items})` : '';
 			return option + this.$locale.baseText('ndv.output.of') + (this.maxRunIndex + 1) + itemsLabel;
 		},
-		getDataCount(
+		getData(
 			runIndex: number,
 			outputIndex: number,
 			connectionType: ConnectionTypes = NodeConnectionType.Main,
-		) {
-			if (this.pinData) {
-				return this.pinData.length;
-			}
+		): INodeExecutionData[] {
+			let inputData: INodeExecutionData[] = [];
 
-			if (this.node === null) {
-				return 0;
+			if (this.node) {
```

Comment:
Or to avoid the unneeded subsequent check `if (inputData.length === 0 || !Array.isArray(inputData))` we can return early:

```suggestion
			if (!this.node) return [];
```

Reply:
I checked git history for this and it was added by Jan to check if the data returned from the code node is an array.